

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>AHA &mdash; Applications 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/pacbio.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="top" title="Applications 1.0 documentation" href="index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li><a href="index.html">Applications 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="aha">
<h1>AHA<a class="headerlink" href="#aha" title="Permalink to this headline">¶</a></h1>
<p>AHA is the hybrid scaffolding approach found in SMRT Analysis. Contigs from a draft assembly generated by a different assembler can be joined using PacBio long reads. The current draft genome size limit for AHA in SMRT Analysis 1.4 is 160 Mbases and 20.000 contigs. On the SMRT Pipe command line fasta files can be used directly for long read coverage. If a mixture of fasta and bas.h5 files is available for PacBio long read coverage it is suggested to export filtered_subreads from the bas.h5 files and combine them with the other fasta file in order to use the following input.xml file.</p>
<p>AHA requires a params.xml and an input.xml.</p>
<div class="section" id="params-xml">
<h2>params.xml<a class="headerlink" href="#params-xml" title="Permalink to this headline">¶</a></h2>
<p>Here is a params.xml:</p>
<div class="highlight-python"><pre>&lt;?xml version="1.0"?&gt;
&lt;smrtpipeSettings&gt;
        &lt;!-- HybridAssembly 1.2.0 parameter file for long reads --&gt;
        &lt;module name="HybridAssembly"&gt;
        &lt;!-- General options --&gt;
        &lt;!-- Parameter schedules are used for iterative hybrid assembly. They are
                given in comma delimited tuples separate by semicolons. The fields
                in order are:

                - Minimum alignment score (aka Z-score). Higher is more stringent.
                - Minimum number of reads needed to link two contigs. (Redundancy)
                - Minimum subread length to participate in alignment.
                - Minimum contig length to participate in alignment.

                If a tuple contains less than 4 fields, defaults will be used for
                the remaining fields. --&gt;
        &lt;paramSchedule&gt;6,3,75;6,3,75;6,2,75;6,2,75&lt;/paramSchedule&gt;

        &lt;!-- Untangling occurs after the main scaffolding step. Valid values
                are "bambus" and "pacbio" (recommended and the default). --&gt;
        &lt;untangler&gt;pacbio&lt;/untangler&gt;

        &lt;!-- Gap fillin can be turned on by setting to True or off by setting to False --&gt;
        &lt;fillin&gt;False&lt;/fillin&gt;

        &lt;!-- These options allow long reads --&gt;
        &lt;longReadsAsStrobe&gt;True&lt;/longReadsAsStrobe&gt;
        &lt;blasrOpts&gt;-minMatch 10 -minPctIdentity 70 -bestn 10 \
        -noSplitSubreads&lt;/blasrOpts&gt;

        &lt;!-- Parallelization options --&gt;
        &lt;numberProcesses&gt;4&lt;/numberProcesses&gt;
        &lt;/module&gt;
&lt;/smrtpipeSettings&gt;</pre>
</div>
</div>
<div class="section" id="input-xml">
<h2>input.xml<a class="headerlink" href="#input-xml" title="Permalink to this headline">¶</a></h2>
<p>The input.xml specifies the contigs and long reads used to scaffold them:</p>
<div class="highlight-python"><pre>&lt;?xml version="1.0"?&gt;
&lt;pacbioAnalysisInputs&gt;
        &lt;dataReferences&gt;

        &lt;!-- High-confidence sequences fasta file --&gt;
        &lt;url ref="assembled_contigs:/home/lhon/analysis/aha/contigs.fasta"/&gt;

        &lt;!-- PacBio reads, either in fasta or in bas.h5 format. --&gt;
        &lt;url ref="file:/home/lhon/analysis/aha/pacbio.filtered_subreads.fasta" /&gt;

        &lt;/dataReferences&gt;
&lt;/pacbioAnalysisInputs&gt;</pre>
</div>
</div>
<div class="section" id="job-sh">
<h2>job.sh<a class="headerlink" href="#job-sh" title="Permalink to this headline">¶</a></h2>
<p>The job.sh will contain the command to run the AHA hybrid assembly module using SMRT Pipe.</p>
<div class="highlight-python"><pre>smrtpipe.py --params=params.xml xml:input.xml</pre>
</div>
</div>
<div class="section" id="running-aha">
<h2>Running AHA<a class="headerlink" href="#running-aha" title="Permalink to this headline">¶</a></h2>
<p>First ensure you have setup the SMRT Analysis environment:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">source</span> <span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">smrtanalysis</span><span class="o">/</span><span class="n">etc</span><span class="o">/</span><span class="n">setup</span><span class="o">.</span><span class="n">sh</span>
</pre></div>
</div>
<p>Afterwards, now run <tt class="docutils literal"><span class="pre">job.sh</span></tt>:</p>
<div class="highlight-python"><pre>source job.sh</pre>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">AHA</a><ul>
<li><a class="reference internal" href="#params-xml">params.xml</a></li>
<li><a class="reference internal" href="#input-xml">input.xml</a></li>
<li><a class="reference internal" href="#job-sh">job.sh</a></li>
<li><a class="reference internal" href="#running-aha">Running AHA</a></li>
</ul>
</li>
</ul>

  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/AHA.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li><a href="index.html">Applications 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2013, Pacific Biosciences.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>