

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>P_ErrorCorrection &mdash; Applications 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/pacbio.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="top" title="Applications 1.0 documentation" href="index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li><a href="index.html">Applications 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="p-errorcorrection">
<h1>P_ErrorCorrection<a class="headerlink" href="#p-errorcorrection" title="Permalink to this headline">¶</a></h1>
<p>P_ErrorCorrection is a module in SMRT Pipe versions 1.3.1 to 1.3.3 that performs error correction on PacBio long reads by mapping shorter, high accuracy reads onto the long reads. The error corrected reads can then be assembled using a long-read assembler, such as Celera Assembler, Mira, or Allora.</p>
<p>If you are using SMRT Analysis 1.4, please consider using the <a class="reference external" href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/HGAP">HGAP approach</a> instead.</p>
<p>Unlike the PacBioToCA module in Celera Assembler, P_ErrorCorrection has the ability to keep long reads together in regions where there is no short read coverage.</p>
<p>P_ErrorCorrection requires a params.xml and an input.xml.</p>
<div class="section" id="params-xml">
<h2>params.xml<a class="headerlink" href="#params-xml" title="Permalink to this headline">¶</a></h2>
<p>P_ErrorCorrection can use fasta/fastq files as input if the options <tt class="docutils literal"><span class="pre">useFastqAsShortReads</span></tt> and <tt class="docutils literal"><span class="pre">useFastaAsLongReads</span></tt> are set to true. The following shows this configuration:</p>
<div class="highlight-python"><pre>&lt;?xml version="1.0"?&gt;
&lt;smrtpipeSettings&gt;
    &lt;module id="P_ErrorCorrection"&gt;
        &lt;description&gt;Error Correction&lt;/description&gt;
        &lt;param name="useFastqAsShortReads" hidden="true"&gt;
            &lt;value&gt;True&lt;/value&gt;
        &lt;/param&gt;
        &lt;param name="useFastaAsLongReads" hidden="true"&gt;
            &lt;value&gt;True&lt;/value&gt;
        &lt;/param&gt;
        &lt;param name="useLongReadsInConsensus" hidden="true"&gt;
            &lt;value&gt;False&lt;/value&gt;
        &lt;/param&gt;
        &lt;param name="useUnalignedReadsInConsensus" hidden="true"&gt;
            &lt;value&gt;False&lt;/value&gt;
        &lt;/param&gt;
        &lt;param name="blasrOpts"&gt;
            &lt;value&gt;-advanceHalf -noSplitSubreads -ignoreQuality -minMatch 10 -minPctIdentity 70 -bestn 20&lt;/value&gt;
        &lt;/param&gt;
        &lt;param name="layoutOpts"&gt;
            &lt;value&gt;--overlapTolerance=25&lt;/value&gt;
        &lt;/param&gt;
    &lt;/module&gt;
&lt;/smrtpipeSettings&gt;</pre>
</div>
<p><tt class="docutils literal"><span class="pre">bas.h5</span></tt> files can also be used directly if either <tt class="docutils literal"><span class="pre">useFastqAsShortReads</span></tt> and <tt class="docutils literal"><span class="pre">useFastaAsLongReads</span></tt> are set to false. The P_Fetch and P_Filter modules are needed in this scenario, and the params.xml needs to be adjusted accordingly; this is detailed in the <a class="reference external" href="http://www.smrtcommunity.com/SMRT-Analysis/Software/software/smrtanalysis/1.3.0/doc/SMRT%20Pipe%20Reference%20Guide.pdf">SMRT Pipe Reference Guide</a>.</p>
<p>Additional parameters are described in the <a class="reference external" href="http://www.smrtcommunity.com/SMRT-Analysis/Software/software/smrtanalysis/1.3.0/doc/SMRT%20Pipe%20Reference%20Guide.pdf">SMRT Pipe Reference Guide</a>.</p>
</div>
<div class="section" id="input-xml">
<h2>input.xml<a class="headerlink" href="#input-xml" title="Permalink to this headline">¶</a></h2>
<p>You can specify the actual data to be error corrected in the <tt class="docutils literal"><span class="pre">input.xml</span></tt> file:</p>
<div class="highlight-python"><pre>&lt;?xml version="1.0"?&gt;
&lt;pacbioAnalysisInputs&gt;
  &lt;dataReferences&gt;
    &lt;url ref="fasta:pacbio.filtered_subreads.fasta" /&gt;
    &lt;url ref="fastq:illumina.fastq" /&gt;
  &lt;/dataReferences&gt;
&lt;/pacbioAnalysisInputs&gt;</pre>
</div>
<p>As the params.xml options suggest, the long reads data need to be in fasta format, and the short read data need to be in fastq format.</p>
<p>If referring to <tt class="docutils literal"><span class="pre">bas.h5</span></tt> files, you can use the following format:</p>
<div class="highlight-python"><pre>&lt;url ref="run:0000000-0000"&gt;&lt;location&gt;/path/to/bas.h5&lt;/location&gt;&lt;/url&gt;</pre>
</div>
<p>The run id needs to be unique. If there are a number of <tt class="docutils literal"><span class="pre">bas.h5</span></tt> files, the input.xml can be autogenerated using the <tt class="docutils literal"><span class="pre">fofnToSmrtpipeInput.py</span></tt> script. This script takes a fofn file, which contains list of filenames separated by carriage returns, and outputs a properly formatted input.xml.</p>
</div>
<div class="section" id="job-sh">
<h2>job.sh<a class="headerlink" href="#job-sh" title="Permalink to this headline">¶</a></h2>
<p>The job.sh will contain the command to run the P_ErrorCorrection module using SMRTPipe.</p>
<div class="highlight-python"><pre>smrtpipe.py --params=params.xml xml:input.xml</pre>
</div>
</div>
<div class="section" id="running-p-errorcorrection">
<h2>Running P_ErrorCorrection<a class="headerlink" href="#running-p-errorcorrection" title="Permalink to this headline">¶</a></h2>
<p>First ensure you have setup the SMRT Analysis environment:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">source</span> <span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">smrtanalysis</span><span class="o">/</span><span class="n">etc</span><span class="o">/</span><span class="n">setup</span><span class="o">.</span><span class="n">sh</span>
</pre></div>
</div>
<p>Afterwards, now run <tt class="docutils literal"><span class="pre">job.sh</span></tt>:</p>
<div class="highlight-python"><pre>source job.sh</pre>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">P_ErrorCorrection</a><ul>
<li><a class="reference internal" href="#params-xml">params.xml</a></li>
<li><a class="reference internal" href="#input-xml">input.xml</a></li>
<li><a class="reference internal" href="#job-sh">job.sh</a></li>
<li><a class="reference internal" href="#running-p-errorcorrection">Running P_ErrorCorrection</a></li>
</ul>
</li>
</ul>

  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/P_ErrorCorrection.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li><a href="index.html">Applications 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2013, Pacific Biosciences.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>