

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>&lt;no title&gt; &mdash; Applications 1.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/pacbio.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="top" title="Applications 1.0 documentation" href="../index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li><a href="../index.html">Applications 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <p>This page describes the use of PreAssembler, Celera® Assembler and Quiver for the purpose of Hierarchical Genome Assembly in SMRT® Analysis version 1.4 with special emphasis on SMRT Portal, the web interface for SMRT Analysis 1.4.</p>
<p>If you are interested in executing the entire workflow on the smrtpipe command line please refer to the HGAP wrapper script developed for that purpose.</p>
<p><a class="reference external" href="https://github.com/PacificBiosciences/Bioinformatics-Training/tree/master/hgap">https://github.com/PacificBiosciences/Bioinformatics-Training/tree/master/hgap</a></p>
<p>HGAP in SMRT Analysis 1.4 has been tested on microbial sized genomes, so while it will run on larger genomes there is no official support for larger-sized genomes.</p>
<p>## Introduction
The Hierarchical Genome Assembly Process (HGAP) was developed to improve the performance, accuracy and efficiency of _de <a href="#id1"><span class="problematic" id="id2">novo_</span></a> assemblies from long single-pass reads generated by the PacBio® sequencing platform. The process consists of 3 main procedures:</p>
<ol class="arabic simple">
<li>Pre-assembly of PacBio single-pass reads;</li>
<li>The assembly of trimmed pre-assembled reads via long read assemblers; and</li>
<li>Polishing the assembly to the highest consensus accuracy via Quiver.</li>
</ol>
<p>## Pre Assembly workflow
The most recent algorithmic advance, HGAP, reduces the sample preparation requirements to a single large insert library. Instead of using just the longest reads in a large insert library (as is the case with error correction), the HGAP approach incorporates almost all the reads. This is accomplished by aligning all subreads above the minimum subread length against the longest seed reads and generating a consensus sequence of the mapped subreads. These pre-assembled reads are both long and high accuracy and can be used by long read assemblers for de novo genome assembly utilizing only PacBio data.</p>
<p>Mapping long subreads to even longer seed reads is able to boost the accuracy of the preassembled reads even in difficult regions of the genome (low complexity, repetitive). SMRT® sequencing’s lack of amplification bias, its ability to span difficult regions of genomes and the random nature of the error model contribute to producing accurate and complete genome assemblies that previously had not been possible or only with substantial investments in manual genome finishing.</p>
<p>### Filtering
After preparing a SMRT®bell insert library that is as long as the quality of the genomic DNA allows (preferably longer than 10kb) and generating single pass reads from the longest movie length possible (currently 120min) the first step before starting preassembly is to evaluate the subread length distribution and cumulative yield starting with the longest subreads.</p>
<p>One would select all the SMRT®cells for the genome to assemble and run the data through the RS_Filter_Only protocol at the minimum RQ (0.8) and minimum subread (500 bp) that will be employed by the RS_PreAssembler protocol in the next step. Given an estimated genome size for the organism of interest one would use the Subread Filtering graph to identify a long subread yield that would correspond to 20-30X genome coverage. For a 5Mb genome one would want to identify the minimum subread cutoff that would generate 100-150Mb coverage which should typically be in the 4kb-6kb range (Fig 1). We will use this subread cutoff as the seed read length for the PreAssembly step.</p>
<p>[[/images/HGAP-Filtering-Plot.png]]</p>
<p>Fig. 1.</p>
<p>### PreAssembly
The RS_PreAssembler protocol utilizes the subreads that are longer than the seed read length cutoff as reference to map all other single pass subreads to their respective <cite>best n</cite> locations. All overlapping subreads that map to a given seed read are then used to generate a consensus sequence which in effect results in a seed read with significantly higher accuracy in regions of high coverage. The stringency and efficiency of the mapping step can be controlled through the advanced choices box that allows the pass through of all BLASR options to the command line (Fig 2).</p>
<p>Prepare the RS_PreAssembler protocol in SMRT®Portal by specifying a <cite>Minimum Seed Read Length</cite> as identified from the subread filtering graph. When using data generated from the XL enzyme in combination with the C2 sequencing kit it is suggested to change the Advanced BLASR option of  <cite>-maxLCPLength</cite> from the default 16 to 14 to accommodate the slightly altered error profile between the C2 and XL polymerase. If one is interested in carrying out a traditional error correction step that utilizes CCS data to map to the seed reads the <cite>Use CCS</cite> box would need to be checked (Fig 2).</p>
<p>[[/images/PreAssembler-SMRT-Portal.png]]</p>
<p>Fig. 2.</p>
<p>### Fastq Trimming
The consensus algorithm of the PreAssembler protocol reports the confidence in the mapped coverage in the form of a Quality Value (QV) Score that is encoded in the <cite>corrected.fastq</cite>. One can therefore use minimum QV filtering to remove regions of low coverage like low quality subreads, seed read edges, merge points of chimeric seed reads and inverted repeats of unidentified adapter sequences.</p>
<p>Choosing automatic fastq filtering in RS_PreAssembler protocol uses a QV cutoff of 59.5 and a length cutoff of 500bp (Fig. 2). If automatic fastq filtering is selected the filtered <cite>corrected.fastq</cite> will replace the original <cite>corrected.fastq</cite> preventing additional filtering with lower stringency to be carried out. You may want to do manual fastq trimming to fine tune filtering to the actual library insert size, total coverage and the specifics of the genome (repeat content, complexity etc.).</p>
<p>Empirical observations suggest that carrying out assemblies at different QV cutoffs (20, 35 or 50) at similar coverage levels (different lengths cutoffs) will help in identifying the best trimming parameters for a given data set. For manual fastq filtering use the <cite>trimFastqByQVWindow.py</cite> script on the command line.</p>
<blockquote>
<div><p>source /opt/smrtanalysis/etc/setup.sh</p>
<p>trimFastqByQVWindow.py -h
Usage: trimFastqByQVWindow.py [&#8211;help] [options] ARG1
Options:
-h, &#8211;help            show this help message and exit
-l LOGFILE,           Specify a file to log to. Defaults to stderr.
-d, &#8211;debug           Increases verbosity of logging
-i, &#8211;info            Display informative log entries
-p, &#8211;profile         Profile this script, dumping to&lt;name&gt;.profile
&#8211;qvCut=QVCUT         quality value cutoff
&#8211;trimFront=TRIMFRONT beginning base pairs to remove(gmin)
&#8211;out=OUT             out file(gdefaults to stdout)
&#8211;fastaOut=FASTAOUT   out fasta file(gdefaults to None)
&#8211;minSeqLen=MINSEQLEN min seq len to keep&gt;</p>
</div></blockquote>
<p>## <strong>Celera Assembly workflow</strong>
The use of Celera Assembler is well documented. In short the first step is the preparation of a <cite>frg</cite> file to deliver the preassembled reads to the Celera Assembler pipeline:</p>
<blockquote>
<div><p>fastqToCA -libraryname reads -technology sanger -reads corrected.fastq &gt; preassembled.frg</p>
<p>Some points to consider:
The -libraryname is required, but isn&#8217;t important for any downstream analysis.
The QV values are offset by 33 by default so the -type sanger parameter can be omitted.
The frg file contains the absolute path of the fastq, so moving either will corrupt the frg file.</p>
</div></blockquote>
<p>The <cite>frg</cite> file is then used to specify the input data for Celera Assemblers <cite>runCA</cite> command:</p>
<blockquote>
<div><p>runCA -p asm -d asm -s asm.spec preassembled.frg &gt; asm.out 2&gt;&amp;1</p>
<p>Some points to consider:
The assembled contigs will be at asm/9-terminator/asm.ctg.fasta.
-p asm specifies the prefix for the results files and can be used without changes.
-d asm specifiesis the output directory.</p>
</div></blockquote>
<p>The <cite>asm.spec</cite> file can be tuned for the compute and cluster environment (see Celera Assembler homepage for details).</p>
<p>## <strong>Quiver workflow</strong>
The quiver consensus algorithm is now the default for the RS_Resequencing protocol in SMRT®Portal 1.4. To improve the consensus accuracy of the assembly one would need to copy the output of the Celera Assembler to the <cite>common/reference_dropbox</cite> to generate a reference in SMRT®Portal. The next step is to carry out the RS_Resequencing protocol either with the same QV and subread filter settings used for Preassembler or less stringent for additional coverage.</p>
<p>The default behavior of the quiver algorithm is geared towards generating the highest quality consensus sequence of the reference – the _de <a href="#id3"><span class="problematic" id="id4">novo_</span></a> assembly in this case – which limits coverage only to unambiguously mapped reads (Fig. 3). For non-repetitive regions with high unique coverage the consensus algorithm is able to achieve accuracies above QV50 but for repetitive regions the consensus accuracy could be reduced due to lower unique coverage or in extreme cases insufficient coverage for Quiver.</p>
<p>[[/images/Assembly-Polishing-SMRT-Portal.png]]</p>
<p>Fig. 3</p>
<p>In regions without sufficient unique coverage the default behavior is to insert _Ns_ into the consensus sequence. One way to combat this behavior is to allow ambiguously mapped long reads to participate in the consensus calculation by deselecting _Use only unambiguously mapped <a href="#id5"><span class="problematic" id="id6">reads_</span></a>. The alternative would be to carry out the default RS_Resequencing protocol and then re-running the <cite>consensus.fastq</cite> export command with an alternate option that will insert the original reference sequence in regions that otherwise would be filled with Ns.</p>
<blockquote>
<div>quiver &#8211;noEvidenceConsensusCall=reference &#8211;algorithm=quiver data/aligned_reads.cmp.h5 -r reference.fasta -o consensus.fastq</div></blockquote>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="../_sources/Use_cases/HGAP.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li><a href="../index.html">Applications 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2013, PacBio.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>